discord/bot.go

```diff
@@ -702,6 +722,20 @@ func (guild *GuildState) handleMessageCreate(s *discordgo.Session, m *discordgo.
 				case Settings:
 					HandleSettingsCommand(s, m, guild, args)
 					return // to prevent the user's message from being deleted
+				case Prompt:
```

Comment:
Same comment as above; highly recommend using the lock for these assignments

Reply:
Didn't see this one before. Should be handled if I understood correctly